packages/legend-application-query/src/stores/QueryEditorStore.ts

```diff
@@ -190,6 +191,19 @@ export class QueryExportState {
     );
   }
 
+  get unallowedPersistInfo(): string | undefined {
+    if (this.persistQueryState.isInProgress) {
+      return 'Query saving is in progress';
+    }
+    if (!this.queryBuilderState.mapping) {
+      return 'Query has no mapping';
+    }
+    if (!(this.queryBuilderState.runtimeValue instanceof RuntimePointer)) {
+      return 'Query has invalid runtime value';
+    }
+    return undefined;
+  }
+
   async persistQuery(createNew: boolean): Promise<void> {
```

Comment:
let's use `generator` function (i.e. `flow` form), also rename this to `createQuery`